common/persistence/client/factory.go

```diff
@@ -75,8 +75,10 @@ type (
 		NewMetadataStore() (p.MetadataStore, error)
 		// NewExecutionStore returns an execution store for given shardID
 		NewExecutionStore(shardID int) (p.ExecutionStore, error)
-		// NewVisibilityStore returns a new visibility store
-		NewVisibilityStore() (p.VisibilityStore, error)
+		// NewVisibilityStore returns a new visibility store,
+		// TODO We temporarily using listClosedExecutionsOrderingByCloseTime to determine whether or not ListByCloseExecutions should
+		// be ordering by CloseTime. This will be removed when implementing https://github.com/uber/cadence/issues/3621
+		NewVisibilityStore(listClosedOrderingByCloseTime bool) (p.VisibilityStore, error)
```

Comment:
maybe `listClosedOrderingByCloseTime` -> `sortByCloseTime` since open workflow doesn't have close time, I think it's clear that close time is for list closed wf.

Reply:
good idea.